### PR TITLE
Allow "!" for breaking change commits

### PR DESCRIPTION
**Pull request by rafaelbubach**
Following conventional commits docs.

[Commit message with ! to draw attention to breaking change](https://www.conventionalcommits.org/en/v1.0.0/#commit-message-with--to-draw-attention-to-breaking-change)

fixes: #186 #185

### CODE REVIEW

File: gitlint/tests/expected/cli/test_cli/test_contrib_1 [hunk 1]
```diff
@@ -1,3 +1,2 @@
 1: CC1 Body does not contain a 'Signed-off-by' line
-1: CT1 Title does not start with one of fix, feat, chore, docs, style, refactor, perf, test, revert, ci, build: "Test tïtle"
```
Comment (rafaelbubach) [inline gitlint/tests/expected/cli/test_cli/test_contrib_1:2]: behavior change: now, if the message is not in the correct format, the validation is ended.

File: gitlint/contrib/rules/conventional_commit.py [hunk 2]
```diff
@@ -23,16 +23,18 @@ class ConventionalCommit(LineRule):
...
-        if not RULE_REGEX.match(line):
+        if not match:
             msg = "Title does not follow ConventionalCommits.org format 'type(optional-scope): description'"
             violations.append(RuleViolation(self.id, msg, line))
+        else:
+            line_commit_type = match.group(1)
+            for commit_type in self.options["types"].value:
+                if line_commit_type == commit_type:
+                    break
+            else:
```
Comment (nedbat) [inline gitlint/contrib/rules/conventional_commit.py:36]:
```suggestion
            if line_commit_type not in self.options["types"].value:
```

**Comment by nedbat** [comment]
Is there something I can do to help get this merged?

**Comment by jorisroovers** [comment]
Sorry for delay, life got in the way. I did quick review, looks good. Didn't test myself, but this is a contrib rule so I'm ok to merge. Would you mind rebasing, I think that should trigged the CI run as well. 
I'm trying to find some time over next few weeks to get a dot release out, we should be able to get this included.

**Comment by rafaelbubach** [comment]
> Sorry for delay, life got in the way. I did quick review, looks good. Didn't test myself, but this is a contrib rule so I'm ok to merge. Would you mind rebasing? I think that should trigger the CI run as well.
> I'm trying to find some time over next few weeks to get a dot release out, we should be able to get this included.

Thanks for answering.
I already did the rebase. I hope that's enough.

**Comment by jorisroovers** [comment]
Some integration tests are failing, please fix :-)

**Comment by jorisroovers** [comment]
CI is doing a gitlint check against your last commit message and that is failing (B6 Body message is missing).

I should take out that gitlint check for PRs, since we end up squashing and changing the commit message anyways - it's annoying that it fails during these PR CI runs. Would you mind fixing your last commit message, I'll fix the CI configuration as part of #221

**Comment by rafaelbubach** [comment]
> CI is doing a gitlint check against your last commit message and that is failing (B6 Body message is missing).
> 
> I should take out that gitlint check for PRs, since we end up squashing and changing the commit message anyways - it's annoying that it fails during these PR CI runs. Would you mind fixing your last commit message, I'll fix the CI configuration as part of #221

Sorry for the delay, I did the correction and forced pushed. Thanks for the warning.

**Comment by jorisroovers** [comment]
Thanks! I'm hoping to do a release next week which will include this.

**Pull request merged**